### DISCUSSION TITLE
node server not working

### DISCUSSION THREAD

**Discussion by ghost** [tutao/tutanota#6548] [Q&A]
- [x] This is not a feature request (existing functionality does not work, **not** missing functionality).
 I will request features on [forum](https://www.reddit.com/r/tutanota/) or via support.
- [x] I've searched and did not find a similar issue.

**Bug in web app**

**Describe the bug**
Running `node server` in the `build` folder does not work.

**To Reproduce**
1. Build web app
2. Go to `build` folder
3. Run `node server`.
4. Does not work.

**Expected behavior**
A web server should start hosting the Tuta web app.

**Screenshots**
```
node:internal/modules/cjs/loader:1152
  throw err;
  ^

Error: Cannot find module '/home/CENSORED/tutanota/build/server'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1149:15)
    at Module._load (node:internal/modules/cjs/loader:990:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:142:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v21.6.1
```

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Browser: Chromium
 - Version: 121.0.6167.160

**Node:**
- Node: 21.6.1
- npm: 10.4.0

**Tuta:**
- Version: 3.122.4

**Additional context**

**Comment by charlag** [comment]
Where should `node server` come from?  Do we have some misleading instructions somewhere?

**Reply by ghost** [reply]
It was found in doc/BUILDING.md. Also using http.server from Python you can’t login due to CORS.

**Reply by PsychotherapistSam** [reply]
@0xC4aE1e5 Make sure you check out a release version, otherwise the master might not be compatible with the API versions from the production environment at tuta.com

**Reply by hasan-alparslan-fidan** [reply]
i am having the exact same problem, any updates @0xC4aE1e5 ?

**Reply by ghost** [reply]
Pretty sure I did.

**Reply by ghost** [reply]
No but I would rather have this problem than not having source code at all for certain things (looking at you, Proton)

**Reply by hasan-alparslan-fidan** [reply]
okay i will update you if i can make this work

**Reply by charlag** [reply]
Sorry, we need to update build instructions. I recommend you do `node make -d prod` to build and run desktop client (in debug mode). You don't need a server then.

**Reply by hasan-alparslan-fidan** [reply]
Is there a way that i can run it on my local web server @charlag  ? and thank you for the response, i will try it after work, i will write the results here

**Reply by charlag** [reply]
If it's locally on your machine then you can just serve the `build` folder with any server and it will work.

If it's on another machine you would have to proxy requests because of CSP
